static/js/lightbox.js

```diff
@@ -15,6 +15,76 @@ let is_open = false;
 // memoized instead of being looked up multiple times.
 const asset_map = new Map();
 
+export class PanZoomControl {
+    // Class for both initializing and controlling the
+    // the pan/zoom functionality.
+    constructor(container) {
+        this.container = container;
+        this.panzoom = panzoom(this.container, {
+            disablePan: true,
+            disableZoom: true,
+            cursor: "auto",
+        });
+
+        // The following events are necessary to prevent the click event
+        // firing where the user "unclicks" at the end of the drag, which
+        // was causing accidental overlay closes in some situations.
+        this.container.addEventListener("panzoomstart", () => {
+            overlays.disable_body_click();
+        });
+
+        this.container.addEventListener("panzoomend", () => {
+            // Don't enable right away or the event will be fired anyway.
+            setTimeout(overlays.enable_body_click, 50);
```

Comment:
I'm not sure I understand this comment.  Also, could it be `setTimeout(x, 0)` (i.e. wait for a paint), or do we need to wait 50ms?

Reply:
I changed it to setTimeout(x, 0) and it still works! I did not know about that, neat.

I also updated the comment. I hope it's more clear now. 

This is the fix for the issue described here: https://github.com/zulip/zulip/issues/18939#issuecomment-880282482 where dragging and releasing the mouse in an area (in this case, the top and bottom margins) where normally clicking would close the lightbox unintentionally.

Reply:
Cool, that makes sense -- and yeah, the `setTimeout(x, 0)` trick is something that one of course prefers not to use, but it is the fix for this precise type of layering problem.